src/tools/debugger/entities/SteppingType.java

```diff
@@ -156,19 +189,29 @@ private static void completeHandleFrameSkip(final Suspension susp) {
       If no tags are given, it is assumed the operation is always valid. */
   public final Class<? extends Tags>[] applicableTo;
 
+  /** Stepping operation is only available when in the dynamic scope of the given entity.
+      If no entity types are given, it is assumed the operation is always valid. */
+  public final EntityType[] inScope;
```

Comment:
It is for operations that are scoped by transactions, or object mutexes, or other dynamic scopes. At the moment, I only have these two examples.